packages/material-ui/src/TextField/TextField.d.ts

```diff
@@ -252,4 +250,6 @@ export type TextFieldClassKey = 'root';
  * - [TextField API](https://material-ui.com/api/text-field/)
  * - inherits [FormControl API](https://material-ui.com/api/form-control/)
  */
-export default function TextField(props: TextFieldProps): JSX.Element;
+export default function TextField<T extends TextFieldVariant = 'standard'>(
```

Comment:
This replaces one issue with another. Generics are lost with higher-order components so any `withRouter`, `connect`, `styled` or `withStyles` will not be able to recover props-by-variant.

Reply:
Yeah, I know. And it looks like a real problem, going to invest a little bit more time in it and try to find the solution

Reply:
Yeah I started commenting before I saw your comment.